crates/hotshot/src/traits/election/static_committee.rs

```diff
@@ -123,44 +106,41 @@ where
         }
     }
 
-    fn default_election_config(
-        num_nodes_with_stake: u64,
-        num_nodes_without_stake: u64,
-    ) -> TYPES::ElectionConfigType {
-        StaticElectionConfig {
-            num_nodes_with_stake,
-            num_nodes_without_stake,
-        }
-    }
-
     fn create_election(
-        entries: Vec<PeerConfig<PUBKEY>>,
-        config: TYPES::ElectionConfigType,
+        mut all_nodes: Vec<PeerConfig<PUBKEY>>,
+        committee_members: Vec<PeerConfig<PUBKEY>>,
         fixed_leader_for_gpuvid: usize,
     ) -> Self {
-        let nodes_with_stake: Vec<PUBKEY::StakeTableEntry> = entries
-            .iter()
-            .map(|x| x.stake_table_entry.clone())
-            .collect();
+        let mut committee_nodes_with_stake = Vec::new();
+        let mut committee_nodes_without_stake = Vec::new();
 
-        let mut committee_nodes_with_stake: Vec<PUBKEY::StakeTableEntry> = Vec::new();
-
-        let mut committee_nodes_without_stake: Vec<PUBKEY> = Vec::new();
-        // filter out the committee nodes with non-zero state and zero stake
-        for node in &nodes_with_stake {
-            if node.get_stake() == U256::from(0) {
-                committee_nodes_without_stake.push(PUBKEY::get_public_key(node));
+        // Iterate over committee members
+        for entry in committee_members
+            .iter()
+            .map(|entry| entry.stake_table_entry.clone())
+        {
+            if entry.get_stake() > U256::from(0) {
+                // Positive stake
+                committee_nodes_with_stake.push(entry);
             } else {
-                committee_nodes_with_stake.push(node.clone());
+                // Zero stake
+                committee_nodes_without_stake.push(PUBKEY::get_public_key(&entry));
             }
         }
-        debug!("Election Membership Size: {}", config.num_nodes_with_stake);
-        // truncate committee_nodes_with_stake to only `num_nodes` with lower index
-        // since the `num_nodes_without_stake` are not part of the committee,
-        committee_nodes_with_stake.truncate(config.num_nodes_with_stake.try_into().unwrap());
-        committee_nodes_without_stake.truncate(config.num_nodes_without_stake.try_into().unwrap());
+
+        // Retain all nodes with stake
+        all_nodes.retain(|entry| entry.stake_table_entry.get_stake() > U256::from(0));
+
+        debug!(
+            "Election Membership Size: {}",
+            committee_nodes_with_stake.len()
+        );
+
         Self {
-            nodes_with_stake,
+            all_nodes_with_stake: all_nodes
```

Comment:
This might be able to be `into_iter`? Since we pass ownership to the function above, we can just consume it here if it's not being used.

Reply:
Sure, I can make this change